chroma-manager/chroma_core/migrations/0037_remove_rsyslog_types.py

```diff
@@ -0,0 +1,1537 @@
+# -*- coding: utf-8 -*-
+import datetime
+import json
+
+from functools import partial
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+from django.contrib.contenttypes.models import ContentType
+from django.db.models import Q
+
+def does_not_contain_content_type_id (content_type_ids, json):
+    return json["locked_item_type_id"] not in content_type_ids
+
+def update_locks_json (content_type_ids, job):
+    locks_json = json.loads(job.locks_json)
```

Comment:
Just a note that you can do pipelining (a-la F#)  with toolz with `pipe`. That way, you don't need to store intermediate values.

You can also import curried versions of `filter`, `map` etc... I usually import them under `cfilter` etc... to avoid overwriting builtins

It would be something like (untested):

```
jobs.locks_json = pipe(
  job.locks_json,
  json.loads,
  cfilter(partial(does_not_contain_content_type_id, content_type_ids)),
  json.dumps
);
```

I don't think it's worth changing right now, just noting it for your reference going forward.

Reply:
I'd like to go ahead and update as it will be pretty quick for me to test since I have a snapshot to work with.